engine/include/logger/instance.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <logger/config.hpp>
+#include <logger/sink.hpp>
+#include <logger/target.hpp>
+#include <memory>
+#include <stdexcept>
+
+namespace gen::logger {
+///
+/// \brief Logger Instance: a single instance must be created within main's scope.
+///
+class Instance {
+public:
+	///
+	/// \brief Error thrown if more than one instance is attempted to be created.
+	///
+	struct DuplicateError : std::runtime_error {
+		using std::runtime_error::runtime_error;
+	};
+
+	Instance(Instance&&) = delete;
+	Instance& operator=(Instance&&) = delete;
+
+	Instance(Instance const&) = delete;
+	Instance& operator=(Instance const&) = delete;
+
+	///
+	/// \brief Create a logger Instance.
+	/// \param filePath path to create/overwrite log file at.
+	/// \param config Config to use.
+	///
+	explicit Instance(char const* filePath = "genesis.log", Config config = {});
+	~Instance();
+
+	///
+	/// \brief Obtain a copy of the Config in use.
+	///
+	[[nodiscard]] Config getConfig() const;
+	///
+	/// \brief Overwrite the Config in use.
+	///
+	void setConfig(Config config);
+
+	///
+	/// \brief Add a custom sink.
+	///
+	void addSink(std::unique_ptr<Sink> sink);
```

Comment:
setConfig and addSink can be made const

Reply:
I don't think we want users to be able to modify the state of `const` instances. The fact that it uses PIMPL is just an implementation detail, changes in state to `Impl` that have public side effects should require a mutable `Instance`.